hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -146,77 +142,28 @@
             enabled, ensure that the raw data is written and read back.
             '''
       milestone: V2
-      tests: []
+      tests: ["flash_ctrl_scramble"]
     }
     {
       name: ecc
       desc: '''
             Randomly enable ECC for a randomly selected set of pages. Randomly corrupt a single bit
-            in the memory that is about to be read and ensure that the ECC works - the corrupted bit
-            should be fixed. Corrupt randomly either the data or the ECC bits. Randomly corrupt 2
-            bits in the same word and ensure that the read results in error. Ensure that pages with
-            ECC not enabled reads back corrupted data without any errors.
-            '''
-      milestone: V2
-      tests: []
-    }
-    {
-      name: read_ecc_metadata
-      desc: '''
-            TBD - in futurem support for reading the metadata from the ECC bits may be supported -
-            this is a placeholder to verify that feature.
-            '''
-      milestone: V2
-      tests: []
-    }
-    {
-      name: error
-      desc: '''
-            This test will attempt to generate flash error in all possible ways:
-            1. Read / program / erase a protected chunk
-            2. Set num words to 0 or out of bounds on read or program
-            3. Inject double bit errors when ECC is enabled
-            4. Program or read out-of-bounds info partition
-            5. Erase whole bank with bank erase not enabled
-            6. Randomly wiggle scramble enable in the middle of an op and ensure that the FSM does
-            not get locked up
-            '''
-      milestone: V2
-      tests: []
-    }
-    {
-      name: flash_macro_timing
-      desc: '''
-            When verifying against the real flash macro (which will be in a closed source repo),
-            check that the read / program / erase timing parameters are correct.
-            '''
-      milestone: V2
-      tests: []
-    }
-    {
-      name: stress_all
-      desc: '''
-            This enables all of the above features to fully stress all the moving parts in the
-            design.
-            '''
-      milestone: V2
-      tests: ["flash_ctrl_stress_all"]
-    }
-    {
-      name: flash_init
-      desc: '''
-            TBD - if supported - verify flash initialization routine upon reset.
+            in the memory that is about to be read and ensure that the ECC works - the corrupted
```

Comment:
this is great. There is one more update inbound (i'm working on it) for another set of ECC.
I will add you to the PR when it is ready and explain more.

Reply:
Ok, this is not top priority test to be developed currently.

Reply:
actually once this change lands, it will impact how you do backdoor writes / reads, so it may have a pretty significant impact to all of your existing tests.  It basically changes the way data is structured in flash.